src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -174,7 +174,7 @@ void QgsLayoutItemLegend::draw( QgsLayoutItemRenderContext &context )
   QgsLegendRenderer legendRenderer( mLegendModel.get(), mSettings );
   legendRenderer.setLegendSize( mSizeToContents ? QSize() : rect().size() );
 
-  legendRenderer.drawLegend( painter );
+  legendRenderer.drawLegend( context );
```

Comment:
Oops, this should be

```suggestion
  legendRenderer.drawLegend( context.renderContext() );
```